cpp/include/cudf/io/json.hpp

```diff
@@ -53,6 +53,69 @@ struct schema_element {
    * @brief Allows specifying this column's child columns target type
    */
   std::map<std::string, schema_element> child_types;
+  // // Add constructors for schema_element to keep order too with initializer list.
+  // // templated constructor with schema_element<bool keep_order=false>
+  // // store the order as
+  /** @brief Allows specifying the order of the columns
+   */
+  std::optional<std::vector<std::string>> column_order;
+
+  /**
+   * @brief Default constructor
+   *
+   * @param type The type that this column should be converted to
+   * @param child_types Allows specifying this column's child columns target type
+   * @param column_order Allows specifying the order of the columns
+   */
+  schema_element(data_type type,
+                 std::map<std::string, schema_element> child_types,
+                 std::optional<std::vector<std::string>> column_order = std::nullopt)
+    : type{std::move(type)},
+      child_types{std::move(child_types)},
+      column_order{std::move(column_order)}
+  {
+  }
+
+  /**
+   * @brief Constructor to create a schema_element from a data_type
+   *
+   * @param type The type that this column should be converted to
+   */
+  schema_element(data_type type) : schema_element{type, {}, std::nullopt} {}
+  schema_element() = default;
```

Comment:
In places where std::map<string, schema_element> dmap;
`dmap["a"] = {data_type{STRING}}` is causing error because `dmap["a"] ` calls default constructor.
I removed all the constructors and kept the interface as simple struct.